### PR TITLE
[mlir][nvvm] Introduce `fence.mbarrier.init`

### PR DESCRIPTION
**Pull request by grypp**
This PR introduce `fence.mbarrier.init` OP

### CODE REVIEW

**Comment by grypp** [comment]
@durga4github can you also give a quick review for this one?

File: mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td [hunk 1]
```diff
@@ -422,6 +422,22 @@ def NVVM_SetMaxRegisterOp : NVVM_PTXBuilder_Op<"setmaxregister"> {
   let hasVerifier = 1;
 }
 
+def NVVM_FenceMbarrierInitOp : NVVM_PTXBuilder_Op<"fence.mbarrier.init"> {
```
Comment (durga4github) [inline mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:425]: I would like to get your thoughts on:
Should we name the Op as "fence.mbarrier_init" ?
(All other places below use the "_" version).

Either ways, change looks good to me.

**Review by durga4github — APPROVED** [review]

File: mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td (hunk 1 above)
Comment (grypp) [reply mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:425]: As far as I can see, MLIR prefers '_' more than '.' . I think I need to change all the OPs to that. I'll do that in follow up PR. It's great that you catch this. So we will have the consistency.

Comment (durga4github) [reply mlir/include/mlir/Dialect/LLVMIR/NVVMOps.td:425]: Sure, thank you!

**Review by durga4github — APPROVED** [review]

**Review by durga4github — APPROVED** [review]

**Pull request merged**